spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpNetServerConnectionFactory.java

```diff
@@ -114,6 +115,7 @@ public void run() {
 						throw new IOException(this + " stopped before accept");
 					}
 					else {
+						publishServerListeningEvent(getPort());
```

Comment:
Is it intentionally that we publish the `TcpConnectionServerListeningEvent` each time when we come here during the main `while (true)` cycle? Each time when we ready to `this.serverSocket.accept()` a new connection from clients.
Looks like `ConnectionFactoryTests.testObtainConnectionIdsNet()` fails with:
```
java.lang.AssertionError: 
Expected: a value equal to or greater than <7>
     but: <5> was less than <7>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:865)
	at org.junit.Assert.assertThat(Assert.java:832)
	at org.springframework.integration.ip.tcp.connection.ConnectionFactoryTests.testObtainConnectionIds(ConnectionFactoryTests.java:158)
	at org.springframework.integration.ip.tcp.connection.ConnectionFactoryTests.testObtainConnectionIdsNet(ConnectionFactoryTests.java:74)
```
just because we don't catch up the next `accept()` cycle.